### ISSUE TITLE
generate:types does not work when payload is included in a server action

### ISSUE THREAD

**Issue by andershermansen** [payloadcms/payload-3.0-demo#186]
I have a client component that I use on login page. From that client component I execute a server action which uses payload. This builds fine and works fine (I see the expected console log when clicking the submit button). But when I execute `pnpm generate:types` to update the types it fails with error about module not found.

If I remove the reference to payload from server action it works again.
Or if I remove the usage of the server action from the included client component it works again.

A simple example put on top of payload-3.0-demo is here: https://github.com/andershermansen/payload-3.0-server-action

```
$ pnpm generate:types

> next-latest-starter@0.1.0 generate:types /Users/anders/code/adsign/payload-3.0-server-action
> payload generate:types


node:internal/process/promises:289
            triggerUncaughtException(err, true /* fromPromise */);
            ^
Error [ERR_MODULE_NOT_FOUND]: Cannot find module '/Users/anders/code/adsign/payload-3.0-server-action/payload' imported from /Users/anders/code/adsign/payload-3.0-server-action/src/components/Action.ts
    at finalizeResolution (node:internal/modules/esm/resolve:264:11)
    at moduleResolve (node:internal/modules/esm/resolve:924:10)
    at defaultResolve (node:internal/modules/esm/resolve:1148:11)
    at nextResolve (node:internal/modules/esm/hooks:750:28)
    at resolve (file:///Users/anders/code/adsign/payload-3.0-server-action/node_modules/.pnpm/payload@3.0.0-beta.24_@swc+core@1.5.0_@swc+types@0.1.6_graphql@16.8.1_typescript@5.4.5/node_modules/payload/dist/bin/loader/index.js:82:12)
    at async nextResolve (node:internal/modules/esm/hooks:750:22)
    at async Hooks.resolve (node:internal/modules/esm/hooks:238:24)
    at async handleMessage (node:internal/modules/esm/worker:199:18) {
  code: 'ERR_MODULE_NOT_FOUND',
  url: 'file:///Users/anders/code/adsign/payload-3.0-server-action/payload'
}

Node.js v21.7.3
 ELIFECYCLE  Command failed with exit code 1.
```

**Comment by andershermansen** [comment]
Reproduction repo at https://github.com/andershermansen/payload-3.0-server-action updated to 3.0.0-beta.29, still same error.

**Comment by kawaunharvey** [comment]
+1 for this. I had to comment out all of my custom components in order to generate the types

**Comment by robertjbass** [comment]
+1 again. Also can't use custom components and generate types

**Comment by jmikrut** [comment]
This is first on our list for tomorrow

**Comment by AlessioGr** [comment]
This has been fixed in 3.0.0-beta.31 by #6341

**Comment by AhmedCoolProjects** [comment]
HI, I have an issue when generating types with payloadcms 3.0 (3.0.0-beta.72), this is the error I get:

$ cross-env NODE_OPTIONS=--no-deprecation payload generate:types

node:internal/process/promises:391
    triggerUncaughtException(err, true /* fromPromise */);
    ^
Error: Bindings not found.
    at Compiler.<anonymous>
...
Node.js v20.16.0
error: script "payload" exited with code 1
